kubernetes/cache/istio.go

```diff
@@ -27,115 +50,733 @@ func (c *kialiCacheImpl) CheckIstioResource(resourceType string) bool {
 }
 
 func (c *kialiCacheImpl) createIstioInformers(namespace string, informer *typeCache) {
-	// Networking API
-	if c.CheckIstioResource(kubernetes.VirtualServices) {
-		(*informer)[kubernetes.VirtualServices] = createIstioIndexInformer(c.istioNetworkingGetter, kubernetes.VirtualServices, c.refreshDuration, namespace)
-	}
+	sharedInformers := istio.NewSharedInformerFactoryWithOptions(c.istioApi, c.refreshDuration, istio.WithNamespace(namespace))
 	if c.CheckIstioResource(kubernetes.DestinationRules) {
-		(*informer)[kubernetes.DestinationRules] = createIstioIndexInformer(c.istioNetworkingGetter, kubernetes.DestinationRules, c.refreshDuration, namespace)
+		(*informer)[kubernetes.DestinationRuleType] = sharedInformers.Networking().V1alpha3().DestinationRules().Informer()
```

Comment:
Would it be possible to save [Informers](https://pkg.go.dev/istio.io/client-go@v1.11.3/pkg/informers/externalversions/networking/v1beta1#Interface) in the namespace cache rather than the `cache.SharedIndexInformer`? Being able to use the listers could simplify a lot of the cache methods to something like:
```
nsCache[kubernetes.DestinationRuleType].Lister().DestinationRules(namespace).Get(name)
```
```
nsCache[kubernetes.DestinationRuleType].Lister().List(selector)
nsCache[kubernetes.DestinationRuleType].Lister().DestinationRules(namespace).List(selector)
```

Here instead of calling
```
(*informer)[kubernetes.DestinationRuleType] = sharedInformers.Networking().V1alpha3().DestinationRules().Informer()
```
you'd save
```
(*informer)[kubernetes.DestinationRuleType] = sharedInformers.Networking().V1alpha3().DestinationRules()
```